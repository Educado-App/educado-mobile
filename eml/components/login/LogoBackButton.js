import React from 'react';
import { View } from 'react-native';
import LeaveButton from '../exercise/LeaveButton';
import EducadoLogo from '../images/EducadoLogo';
import PropTypes from 'prop-types';


/**
 * Component that includes, logo, title and backbutton, used in login and register screens
 * @param {Object} props Should contain the following properties:
 * - navigationPlace: String
 * @returns {React.Element} Header/logo/back button component

 */
export default function LogoBackButton(props) {

	return (
		<View className='flex-row justify-center items-center w-full mt-4'>
			{/* TODO: Implement with general back button instead */}
			<View className='absolute left-0 z-50'>
				<LeaveButton
					navigationPlace={props.navigationPlace ? props.navigationPlace : 'Home'}
				/>
			</View>
			{/* Educado logo */}
			<View className='items-center justify-center w-full'>
				<EducadoLogo />
			</View>
		</View>
	);
}

LogoBackButton.propTypes = {
	navigationPlace: PropTypes.string,
};