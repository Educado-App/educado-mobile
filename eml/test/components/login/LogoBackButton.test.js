import renderer from 'react-test-renderer';
import React from 'react';
import LogoBackButton from '../../../components/login/LogoBackButton';
import { useNavigation } from '@react-navigation/native';

// Mock the useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

test('Ensure that the field is rendered correctly without props', async () => {
  let logoBackButton;
  await renderer.act(() => {
    logoBackButton = renderer.create(<LogoBackButton />);
  });
  let tree = logoBackButton.toJSON();
  expect(tree).toMatchSnapshot();
});
