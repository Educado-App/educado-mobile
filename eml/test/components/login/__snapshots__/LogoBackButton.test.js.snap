// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the field is rendered correctly without props 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
      {
        "width": "100%",
      },
      {
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "position": "absolute",
        },
        {
          "left": 0,
        },
        {
          "zIndex": 50,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 20,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "invisible",
              "borderColor": "#2089dc",
              "borderRadius": 20,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "padding": 4,
              "paddingHorizontal": 6,
              "width": 50,
            }
          }
        >
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#383838",
                        "fontSize": 25,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  Û∞ÖÅ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        [
          {
            "height": 48,
          },
        ]
      }
    />
  </View>
</View>
`;
